src/game.py

```diff
@@ -1,52 +1,38 @@
+import src.constants as constants
+
 from src.cli import Cli
-from src.board import Board
 from src.player import Player
+from src.validator import Validator
 
-class Game():
-    def __init__(self, cli=Cli(), board=Board(), players=[Player(token="X"),
-                                                          Player(token="O")]):
+class Game:
+    def __init__(self, cli, players, board):
         self.cli = cli
-        self.board = board
         self.players = players
-        self.exit = False
-
-    def start(self):
-        self.cli.welcome()
-        self.play()
+        self.board = board
+        self.game_in_process = True
+        self.possible_results = self.cli.build_possible_results(self.players)
+        self.actions = {
+            constants.MOVE: (lambda move, token: self.board.move_result(move, token)),
+            constants.EXIT: (lambda move, token: constants.EXIT),
+            constants.ERROR: (lambda move, token: self.cli.invalid_move()) 
+        } #contains lambdas
 
     def play(self):
-        self.cli.display_board(self.board)
-        while not self.game_is_over():
-            self.turn()
-        self.handle_exit()
-
-    def handle_exit(self):
-        if self.exit:
-            self.cli.handle_exit()
-        else:
-            self.cli.handle_game_end()
-
-    def game_is_over(self):
-        return self.exit or self.board.full()
-
-    def turn(self):
-        while True:
-            player_choice = self.cli.prompt_player_turn(self.current_player())
-            if player_choice == self.cli.EXIT:
-                self.exit = 1
-                break;
-            if self.cli.validate_input(player_choice) and self.board.valid_move(self.input_to_index(player_choice)):
-                move = self.input_to_index(player_choice)
-                break;
-            self.cli.invalid_move()
-        if not self.exit:
-            self.board.update(move, self.current_player().token)
-            self.cli.display_board(self.board)
-       
-    def input_to_index(self, user_input):
-        return int(user_input) - 1 
-
+        result = None
+        validator = Validator()
+        self.cli.print_board(self.board)
+        while(self.game_in_process):
+            current_player = self.current_player()
+            player_choice = self.cli.request_move(current_player)
+            selected_action, move = validator.validate_move(player_choice, self.board)
+            result = self.actions[selected_action](move, current_player.token)
+            if result != constants.EXIT: self.cli.print_board(self.board)
+            if result in self.possible_results: self.game_in_process = False
```

Comment:
It is helpful to pull these boolean expressions into helper functions sometimes. The name of the helper function helps readability and drives home the intent of the check. I would say both if statements could use that refactor.